### PR TITLE
[NOD-1566] Add caching to all stores

### PR DESCRIPTION
**Pull request by stasatdaglabs**

### CODE REVIEW

File: domain/consensus/datastructures/pruningstore/pruningstore.go [hunk 1]
```diff
@@ -15,14 +15,13 @@ var pruningSerializedUTXOSetkey = dbkeys.MakeBucket().Key([]byte("pruning-utxo-s
 type pruningStore struct {
 	pruningPointStaging      *externalapi.DomainHash
 	serializedUTXOSetStaging []byte
+	pruningPointCache        *externalapi.DomainHash
+	serializedUTXOSetCache   []byte
```
Comment (someone235) [inline domain/consensus/datastructures/pruningstore/pruningstore.go:19]: I don't think we want this cache

File: domain/consensus/utils/lrucache/lrucache.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+// Remove removes the entry for the the given key. Does nothing if
+// the entry does not exist
+func (c *LRUCache) Remove(key *externalapi.DomainHash) {
+	delete(c.cache, *key)
+}
+
+func (c *LRUCache) evictRandom() {
+	var keyToEvict externalapi.DomainHash
+	for key := range c.cache {
+		keyToEvict = key
```
Comment (someone235) [inline domain/consensus/utils/lrucache/lrucache.go:55]: You need to break here

File: go.sum [hunk 3]
```diff
@@ -58,6 +56,8 @@ github.com/jrick/logrotate v1.0.0 h1:lQ1bL/n9mBNeIXoTUoYRlK4dHuNJVofX9oWqBtPnSzI
 github.com/jrick/logrotate v1.0.0/go.mod h1:LNinyqDIJnpAur+b8yyulnQw/wDuN1+BYKlTRt3OuAQ=
 github.com/kaspanet/go-secp256k1 v0.0.2 h1:KZGXddYHxzS02rx6EPPQYYe2tZ/rREj4P6XxgQQwQIw=
 github.com/kaspanet/go-secp256k1 v0.0.2/go.mod h1:W9OcWBKzH8P/PN2WAUn9k2YmZG/Uc660WAL1NTS3G3M=
+github.com/kaspanet/golang-lru v0.5.6 h1:2gQxKfeXw+OyXk8WBUr69Rr0Fl/FTRI5RP+LE2hSLv4=
```
Comment (someone235) [inline go.sum:59]: This is not needed

**Review by someone235 — CHANGES_REQUESTED** [review]

File: domain/consensus/datastructures/pruningstore/pruningstore.go (hunk 1 above)
Comment (stasatdaglabs) [reply domain/consensus/datastructures/pruningstore/pruningstore.go:19]: We spoke about this and decided to keep it for now.

**Review by someone235 — APPROVED** [review]

**Review by someone235 — APPROVED** [review]

**Pull request merged**